### PR TITLE
Add BL0940 documentation

### PR DESCRIPTION
**Pull request by tobias-**
## Description:

This adds documentation for the new BL0940 integration

**Related issue (if applicable):** fixes esphome/feature-requests#770

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1904

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.
  - [X] Added images

### CODE REVIEW

**Comment by tobias-** [comment]
I can't figure out how to generate the images needed, so if someone could please help me with that. I copied with pride from pzem004t documentation, and that integration has a `images/pzem004t.svg`.

**Comment by jesserockz** [comment]
As this is a new feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase if needed

**Comment by tobias-** [comment]
@jesserockz fixed

**Comment by tobias-** [comment]
@jesserockz Added image as well. Like I said, graphical stuff is not one of my strengths

File: components/sensor/bl0940.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    :image: bl0940.png
+    :keywords: BL0940
+
+.. note::
+
+    This page is incomplete and could some work. If you want to contribute, please read the
+    :doc:`contributing guide </guides/contributing>`. This page is missing:
+
+      - An image for the front page.
+      - Images/screenshots/example configs of this device being used in action.
```
Comment (oxan) [inline components/sensor/bl0940.rst:15]: I don't think this note is necessary anymore.

File: components/sensor/bl0940.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+The ``bl0940`` sensor platform allows you to use BL0940 energy monitors sensors with
+ESPHome. These are used in some Tuya-devices (e.g. the power metering BW-SHP10)
```
Comment (oxan) [inline components/sensor/bl0940.rst:18]:
```suggestion
ESPHome. These are used in some Tuya-devices (e.g. the power metering BW-SHP10).
```

File: components/sensor/bl0940.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+------------------------
+
+- **voltage** (*Optional*): The voltage value of the sensor in Volts.
+  All options from :ref:`Sensor <config-sensor>`.
+- **current** (*Optional*): The current value of the sensor in Amperes. All options from
+  :ref:`Sensor <config-sensor>`.
+- **power** (*Optional*): The (active) power value of the sensor in Watts. All options from
+  :ref:`Sensor <config-sensor>`.
+- **energy** (*Optional*): Use the voltage value of the sensor in kWh.
+  All options from :ref:`Sensor <config-sensor>`.
```
Comment (oxan) [inline components/sensor/bl0940.rst:61]:
```suggestion
- **energy** (*Optional*): The cumulative energy measured by the sensor in kWh.
  All options from :ref:`Sensor <config-sensor>`.
```

**Review by oxan — APPROVED** [review]

**Pull request closed without merging**